src/methods/get.js

```diff
@@ -554,22 +554,21 @@ export function GetTemplateObject(realm: Realm, templateLiteral: BabelNodeTempla
 }
 
 export function GetFromArrayWithWidenedNumericProperty(realm: Realm, arr: ArrayValue, P: string | SymbolValue): Value {
-  let type = Value;
-
   if (typeof P === "string") {
-    // these are safe methods to allow, as they either return a new array or are a method
-    // that doesn't mutate global state or affect that of the values passed in as arguments
-    if (P === "map" || P === "slice" || P === "filter" || P === "concat" || P === "indexOf" || P === "reverse") {
-      let proto = arr.$GetPrototypeOf();
-      invariant(proto instanceof ObjectValue);
-      return OrdinaryGet(realm, proto, P, arr);
-    }
+    let proto = arr.$GetPrototypeOf();
+    invariant(proto instanceof ObjectValue);
+    let value = OrdinaryGet(realm, proto, P, arr);
```

Comment:
This can run a getter. Since we don't know what P is, can we risk running it here? It might be worthwhile to write a test case that tries to break this.

Reply:
I'll move it to the respective block below.

Reply:
P is a string, but proto[P] might run a getter.